### PR TITLE
Improve module commands

### PR DESCRIPTION
**Pull request by tarsius**
@npostavs This is still very much work-in-progress and completely untested and undocumented, but gives a pretty good idea of I intend to do. This succeeds your earlier drafts; do you agree this is the way forward?

### CODE REVIEW

**Comment by tarsius** [comment]
An earlier discussion can be found in #2674.

**Comment by tarsius** [comment]
@npostavs I am going to delete `np/submodules` and `np/submodules-v2` in a few days, so if you want to hold on to something in there you should make sure you have those branches locally.

**Comment by npostavs** [comment]
Hi, sorry I keep losing steam on this, I don't actually use submodules often enough to stay motivated I guess.

> This succeeds your earlier drafts; do you agree this is the way forward?

If I'm reading this right, the main change is move the submodule update method from a prompt to a popup switch. That looks like a good idea.

Putting all those actions in the submodule popup's top level looks a bit overwhelming to me though.

**Comment by tarsius** [comment]
> Putting all those actions in the submodule popup's top level looks a bit overwhelming to me though.

Right. That's the main reason I have been keeping it back ;-)

**Comment by tarsius** [comment]
I think this is a dead end. And unfortunately my ideas for alternative approaches are rather vague at the moment.

**Comment by npostavs** [comment]
I'm in the process of removing the submodules from yasnippet, and it seems there is no way to do this from magit. The answers in https://stackoverflow.com/questions/1260748/how-do-i-remove-a-submodule make it seem complicated, though possibly simpler in more recent git versions.

**Comment by fbergroth** [comment]
What about putting all the `magit-submodule-*-all` under another popup, say, `magit-submodule-foreach-popup`. The same lowercase keys can be used as their corresponding "one module" variant. Then binding a key, say, `F` (as in "Foreach module...") from `magit-submodule-popup` to `magit-submodule-foreach-popup`.

**Comment by tarsius** [comment]
Thanks for the suggestion. No comment for now - but I intend to look into this in about a week.

File: lisp/magit-submodule.el [hunk 1]
```diff
@@ -158,41 +181,67 @@ it is nil, then PATH also becomes the name."
...
 ;;;###autoload
-(defun magit-submodule-init ()
-  "Register all remaining modules."
-  (interactive)
+(defun magit-submodule-register (modules)
+  "Register MODULES."
+  ;; This command and the underlying "git submodule init" do NOT
+  ;; "initialize" modules.  They merely "register" modules in the
+  ;; super-projects $GIT_DIR/config file, the purpose of which is to
+  ;; allow users to change such values before actually initializing
```
Comment (npostavs) [inline lisp/magit-submodule.el]: Wouldn't it be better to have this explanation in the docstring, rather than hiding it away in source code comments?

Comment (tarsius) [reply lisp/magit-submodule.el]: I plan to come back to this eventually, but considering for how long I have been dragging this out I am pretty happy I finally merged something. I didn't want to further delay this by polishing documentation for who knows how long, so this is why I have put that in a comment for the time being. 😜

Comment (npostavs) [reply lisp/magit-submodule.el]: Hah, fair enough. :)

**Pull request merged**